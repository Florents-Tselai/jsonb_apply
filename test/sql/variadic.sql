select jsonb_apply_variadic('"hello"', 'lower(text)', NULL);
select jsonb_apply_variadic('"hello"', 'upper(text)', NULL);
select jsonb_apply_variadic('"hello"', 'reverse(text)', NULL);

select jsonb_apply_variadic('{"f": "John", "l": "Doe", "message": "Who are you?", "arr": ["Hello", {"k": "value"}]}', 'lower(text)', NULL);
select jsonb_apply_variadic('{"f": "John", "l": "Doe", "message": "Who are you?", "arr": ["Hello", {"k": "value"}]}', 'upper(text)', NULL);
select jsonb_apply_variadic('{"f": "John", "l": "Doe", "message": "Who are you?", "arr": ["Hello", {"k": "value"}]}', 'reverse(text)', NULL);


select jsonb_apply_variadic('"hello world"', 'replace(text, text, text)', 'hello', 'bye');

select jsonb_apply_variadic('"Pg"', 'repeat(text, integer)', 3);

select jsonb_apply_variadic('"abc~@~def~@~ghi"', 'split_part(text,text,integer)', '~@~', 2);
