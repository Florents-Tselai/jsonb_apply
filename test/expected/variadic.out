select jsonb_apply_variadic('"hello world"', 'replace(text, text, text)', '"hello"', '"bye"');
 jsonb_apply_variadic 
----------------------
 "bye world"
(1 row)

select jsonb_apply_variadic('"Pg"', 'repeat(text, integer)', 2);
 jsonb_apply_variadic 
----------------------
 "bye world"
(1 row)

select jsonb_apply_variadic('"abc~@~def~@~ghi"', 'split_part(text,text,integer)', '~@~', 2);
 jsonb_apply_variadic 
----------------------
 "bye world"
(1 row)

